sdk/dotnet/Pulumi/Serialization/OutputCompletionSource.cs

```diff
@@ -75,7 +75,7 @@ internal static class OutputCompletionSource
             {
                 var propType = prop.PropertyType;
                 var propFullName = $"[Output] {resource.GetType().FullName}.{prop.Name}";
-                if (!propType.IsConstructedGenericType &&
+                if (!propType.IsConstructedGenericType ||
```

Comment:
Is this intentional?  It doesn’t look right.

Reply:
This is an intentional fix for a bug which got exposed while testing this PR. The idea is `not(a & b)` so it should be (not a) or (not b)